data-otservbr-global/scripts/actions/quests/feaster_of_souls/bosses_killed.lua

```diff
@@ -5,24 +5,21 @@ local bosses = {
 	["the pale worm"] = { storage = Storage.Quest.U12_30.FeasterOfSouls.PaleWormKilled },
 }
 
-local bossesFeasterOfSouls = CreatureEvent("FeasterOfSoulsKill")
-function bossesFeasterOfSouls.onKill(creature, target)
-	local targetMonster = target:getMonster()
+local bossesFeasterOfSouls = CreatureEvent("FeasterOfSoulsBossDeath")
+function bossesFeasterOfSouls.onDeath(creature)
+	local targetMonster = creature:getMonster()
 	if not targetMonster or targetMonster:getMaster() then
 		return true
 	end
 	local bossConfig = bosses[targetMonster:getName():lower()]
 	if not bossConfig then
 		return true
 	end
-	for key, value in pairs(targetMonster:getDamageMap()) do
-		local attackerPlayer = Player(key)
-		if attackerPlayer then
-			if bossConfig.storage then
-				attackerPlayer:setStorageValue(bossConfig.storage, 1)
-			end
+	onDeathForDamagingPlayers(creature, function(creature, player)
+		if bossConfig.storage then
```

Comment:
the variable storage, doesn't exists. and every time is enabled to kill boss.. 
we need to change this check to use  `if not player:canFightBoss(BOSS_NAME)`